src/user-data.ts

```diff
@@ -169,110 +174,81 @@ export type AppToggleableRequirementChoices = Readonly<Record<string, string>>;
  * Creates credit range based on course
  * Example: [1, 4] is the credit range for the given course
  */
-const createCourseCreditRange = (course: FirestoreSemesterCourse): readonly [number, number] => {
+const createCourseCreditRange = (course: CornellCourseRosterCourse): readonly [number, number] => {
   const courseCreditRange: number[] = [];
-  if (typeof course.creditRange !== 'undefined') {
-    return course.creditRange;
-  }
-  if (typeof course.enrollGroups !== 'undefined') {
-    course.enrollGroups.forEach(enrollGroup => {
-      courseCreditRange.push(enrollGroup.unitsMinimum);
-      courseCreditRange.push(enrollGroup.unitsMaximum);
-    });
-    return [Math.min(...courseCreditRange), Math.max(...courseCreditRange)];
-  }
-  // @ts-ignore
-  return [course.credits, course.credits];
+  course.enrollGroups.forEach(enrollGroup => {
+    courseCreditRange.push(enrollGroup.unitsMinimum);
+    courseCreditRange.push(enrollGroup.unitsMaximum);
+  });
+  return [Math.min(...courseCreditRange), Math.max(...courseCreditRange)];
 };
 
-export const firestoreCourseToAppCourse = (
-  course: FirestoreSemesterCourse,
+export const cornellCourseRosterCourseToAppCourse = (
+  course: CornellCourseRosterCourse,
   isRequirementsCourse: boolean,
   incrementID: () => number,
   addColor: (subject: string) => string
 ): AppCourse => {
-  const uniqueID = course.uniqueID || incrementID();
-
-  const subject = (course.code && course.code.split(' ')[0]) || course.subject;
-  const number = (course.code && course.code.split(' ')[1]) || course.catalogNbr;
-
-  // TODO: same field?
-  const name = course.titleLong || course.name;
+  const uniqueID = incrementID();
```

Comment:
Even though `incrementID()` is used for `cornellCourseRosterCourseToAppCourse`, since it is decoupled from `firestoreCourseToAppCourse` now, would we still have to worry about `incrementID()` changing user data as mentioned in #217?

Reply:
Yes, since all those functions are still calling the roster conversion one